### DISCUSSION TITLE
Pinia custom plugin gets untyped

### DISCUSSION THREAD

**Discussion by umshakalaka** [vuejs/pinia#1816] [Help and Questions]
Hi all!

Here I'm using $api plugin which is a class and has some functions like get, pust...

import { $api } from '@/services/api'
import storage from 'pinia-plugin-persistedstate'

```ts
export function apiPlugin({ store }: PiniaPluginContext) {
  store.$api = $api
}
createPinia().use(storage).use(apiPlugin)


const store = useDevicesStore()
console.log(store.$api)
```

store.$api -> getting red in VS code and not getting typed even though i can see i have $api in the console

Any help, suggestions will be apreciated!
Thank you

**Comment by posva** [comment]
https://pinia.vuejs.org/core-concepts/plugins.html#typing-new-store-properties
